pkg/controller/elasticsearch/pdb/reconcile_with_roles_test.go

```diff
@@ -1014,38 +915,43 @@ func Test_allowedDisruptionsForRole(t *testing.T) {
 			want: 1,
 		},
 		{
-			name: "green health but only 1 master: 0 disruptions allowed for master role",
+			name: "single-node cluster (not high-available): 1 disruption allowed for master role when the cluster is yellow.",
+			args: args{
+				es:              esv1.Elasticsearch{Status: esv1.ElasticsearchStatus{Health: esv1.ElasticsearchYellowHealth}},
+				role:            []esv1.NodeRole{esv1.MasterRole},
+				allStatefulSets: sset.StatefulSetList{ssetfixtures.TestSset{Replicas: 1, Master: true, Data: true}.Build()},
+			},
+			want: 1,
+		},
+		{
+			name: "green health but only 1 master: 1 disruption allowed as the cluster is non-ha",
```

Comment:
I think test description does not match the test here:
* test is for data roles
* data tier is actually HA
* cluster is green anyway?